integration-tests/src/test_loop/tests/state_sync.rs

```diff
@@ -356,11 +380,40 @@ fn slow_test_state_sync_current_epoch() {
                 .iter()
                 .map(|(shard_id, produced)| (*shard_id, produced.to_vec()))
                 .collect(),
+            t.skip_sync_block,
         );
         run_test(state);
     }
 }
 
+// Here we drop the block that's supposed to be the sync hash after the first full epoch,
+// which causes it to be produced but then skipped on the final chain. If the state sync code
+// is unaware of the possibility of forks, this will cause the producer of that block to
+// believe that that block should be the sync hash block, while all other nodes will
+// believe it should be the next block.
+#[test]
+#[ignore]
+fn test_state_sync_forks() {
+    let params =     StateSyncTest {
+        num_validators: 4,
+        num_shards: 5,
+        generate_shard_accounts: true,
+        chunks_produced: &[],
```

Comment:
Why no chunks?

Reply:
Ah this is maybe a little unclear, but this means produce all chunks. The testloop code will check the bools in this slice to decide whether to produce chunks, and for chunk heights greater than the len() of this slice, it will always produce them